types/index.test.ts

```diff
@@ -213,4 +213,42 @@ function* mySaga(): Effects.SagaGenerator<void> {
       return "hello";
     }),
   });
+
+  function outer<T>(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    emit: (item: T) => T,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    handler: (item: T) => T,
+  ) {}
+  const obj = { outer };
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  function makeEmit<T>(x: T, y: T) {
+    return (item: T) => item;
+  }
+
+  function handler(item: number) {
+    return item;
+  }
+  const emitter = makeEmit<number>(2, 3);
```

Comment:
Contrived fake functions to exercise the kind of double-generic situation where the current inference falls down.